src/shared/injector.js

```diff
@@ -0,0 +1,139 @@
+/* global Map */
+
+/**
+ * @typedef Provider
+ * @prop {any} [value] - The value for the object
+ * @prop {Function} [class] - A class that should be instantiated to create the object
+ * @prop {Function} [factory] - Function that should be called to create the object
+ */
+
+/**
+ * @param {Provider} provider
+ */
+function isValidProvider(provider) {
+  if (typeof provider !== 'object' || provider === null) {
+    return false;
+  }
+
+  return (
+    'value' in provider ||
```

Comment:
Just to sanity-check: value registration...is new in this set of changes?

Reply:
Yes. Previously values were handled by using `register("something", () => aValue)`. That doesn't work if the default handling of a function is to treat it like a class, because a quirk of constructors is that if they return an _object_ that is the result of the constructor, but if they return a primitive, it is ignored.

Reply:
But it _would_ work if the default `provider` was assumed to be a function reference instead of a class :)